jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/migration/MigrationManager.java

```diff
@@ -389,6 +389,9 @@ private void upgradeProcessInstance(KieRuntime oldkruntime,
         if (process == null) {
             report.addEntry(Type.ERROR, "Could not find process " + processId);
         }
+        if (processInstance == null) {
+            return;
+        }
```

Comment:
Is this change needed for the upgrade? In case we want to avoid NPE, I would suggest to check if there is any error in the report and if so, just return. This is just a partial solution because if `process` or `processId` is null, then we will continue. So we should stop in case any of the three is null.

In case it is not needed for the upgrade, I would solve it in a separate PR.

Reply:
Hi @MarianMacik, I remember that I've got a complaint from the building process due to the lack of a null check there. If `processInstance` is null, then the next line will break (`processInstance.getProcessId().equals(processId)`).
Anyway, I removed locally this check and the build passed so I'll revert this change. Let's see if we get a green build from Jenkins.